test.c

```diff
@@ -23,44 +22,61 @@ int main() {
   char *string2 = "A string declared as a pointer.\n";
 
   /* Example 3 */
-  char string3[128];   // can copy in up to 127 characters
-                       // chose 128 because it is a multiple of 8 bytes
+  char string3[128]; // can copy in up to 127 characters
+                     // chose 128 because it is a multiple of 8 bytes
   strcpy(string3, "A string constant copied in.\n");
 
   // You can uncomment these to see how the strings were initialized above.
-  // Then you can comment these out again once you see that the 
+  // Then you can comment these out again once you see that the
   // strings were created correctly
   // printf ("%s", string1);
   // printf ("%s", string2);
   // printf ("%s", string3);
 
   // -------------------------------------------------------------
-  // Thorough and exhaustive tests of hasUniqueChars()  
-  bool ok;    // holds the result of hasUniqueChars(); used in asserts
-  
+  // Thorough and exhaustive tests of hasUniqueChars()
+  bool ok; // holds the result of hasUniqueChars(); used in asserts
+
+  // Test Characters:
+
   // Test 1: a test case that should return false because characters are duplicated
-  strcpy(string3, "This should fail (l and s and i and h)");
-  ok = hasUniqueChars(string3);
-  assert(!(ok));
-  
+  // strcpy(string3, "This should fail (l and s and i and h)");
+  // ok = hasUniqueChars(string3);
+  // assert(!(ok));
+
   // Test 2: This should be true and not fail, but won't work until you
   // complete the hasUniqueChars() function
   strcpy(string3, "abcdefghij");
   ok = hasUniqueChars(string3);
   assert(ok);
-  
-  // TODO: add your tests here
-
-
 
   // NOTE: if you add a \n to any tests, this should cause the
-  //       exit failure given. Keep this test last so you can check 
+  //       exit failure given. Keep this test last so you can check
   //       that others pass or fail as you expect them to.
   strcpy(string3, "a b cc\n");
   ok = hasUniqueChars(string3);
+  assert(ok);
+
+  strcpy(string3, "abbc");
+  assert(!hasUniqueChars(string3));
+
+  strcpy(string3, "aABbbBBB");
+  assert(!hasUniqueChars(string3));
+
+  strcpy(string3, "aBcD");
+  assert(!hasUniqueChars(string3));
+
+  strcpy(string3, "abcdefghijklmnopqrstuvwxyz");
+  assert(!hasUniqueChars(string3));
+
+  strcpy(string3, "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ");
+  assert(!hasUniqueChars(string3));
```

Comment:
check non-alpha characters, you never checked the second set of characters, or punctuation etc.